packages/core/src/components/forms/_file-upload.scss

```diff
@@ -69,21 +70,21 @@ $file-upload-button-width-large: $pt-grid-size * 8.5;
   top: 0;
   right: 0;
   left: 0;
-  margin-right: $file-upload-button-width;
   border-top-right-radius: 0;
   border-bottom-right-radius: 0;
   height: $pt-input-height;
+  padding-right: 0;
   line-height: $pt-input-height;
   color: $pt-text-color-muted;
   user-select: none;
 
-  &::after {
+  // we use ::before rather than ::after although the "Browse" button is visually
+  // positioned to the right since this allows for .pt-file-upload-input to be
+  // paired with .pt-text-overflow-ellipsis if desired.
+  &::before {
     @include pt-button();
-    position: absolute;
-    top: 0;
-    right: 0;
-    left: 100%;
-    margin-left: -$button-border-width;
+    float: right;
```

Comment:
I suppose that in certain conditions, the floating button could wrap. Even though some early components still use float, we try to do layout with flex as much as possible now, which we've found more robust